docs/design/kademlia_audit_gating.md

```diff
@@ -0,0 +1,94 @@
+
+## Background
+
+### White paper: 4.6.1 Mitigating Sybil attacks
+While we’ve adopted the proof-of-work scheme S/Kademlia proposes to partially address Sybil attacks, we extend Kademlia with an application specific integration to further defend our network. 
+
+
+Given two storage nodes, A and B, storage node B is not allowed to enter storage node A’s routing table until storage node B can present a signed message from a Satellite C that storage node A trusts claiming that B has passed enough audits that C trusts it (sections 4.13 and 4.15). This ensures that only nodes with verified disk space have the opportunity to participate in the routing layer.
+
+
+A node that is allowed to enter routing tables is considered vetted and lookups only progress through vetted nodes. To make sure unvetted nodes can still be found, vetted nodes keep unbounded lists of their unvetted neighbors provided that the XOR distance to all unvetted neighbors is no farther than the farthest of the k-closest vetted neighbors. Unvetted nodes keep their k-nearest vetted nodes up-to-date.
+
+
+## Goals
+1. Add a signed message from Satellites to authenticate whether a Node has a high enough ID generation difficulty and disk space.
+
+2. Create a trusted Satellite list that contains the IDs of Satellites from which Nodes will accept verification signatures of other Nodes.
+
+3. Implement a Routing Table Antechamber: An XOR-ordered data structure in which unverified nodes are entered if successfully contacted. 
+
+4. Modify FindNear to return n XOR-closest nodes from the antechamber in addition to those already returned from the routing table.
+
+5. On deployment, avoid complete erasure of the network's routing tables.
+
+
+## Terminology
+**Routing Table Antechamber** - *XOR-ordered temporary holding place for unverified storagenodes*
+
+**Node Neighborhood** - *The k-closest nodes to self where distance is measured by XOR. A node is within the node neighborhood if it is closer than the furthest node in the neighborhood. The vetted node neighborhood is the k-closest nodes that are currently in the Routing Table.*
+
+[TODO Node Neighborhood Diagram]
+
+## Design
+
+1. Satellite Signatures for Node Verification
+    - Identities can sign messages already
+    - Create a Message (protobuf) to sign that Satellite C says Node B has been vetted
+    - Get info about the Node's difficulty and disk space to add to the message
+
+2. Trusted Satellites List
+    - Create Whitelist/blacklist with an abstraction layer for trusted/untrusted Satellites
+    - These lists will live on each Node
+
+3. Routing Table Antechamber
+    - XOR ordered data structure (perhaps an ordered slice)
+    - A node can be added if it would be within the vetted node neighborhood
+    - Once a node has been verified, it broadcasts its new status to try to join routing tables
```

Comment:
how does it broadcast its status to join routing tables? also, once a node is in an antechamber, how does its status change to verified? is it periodically re-checked by the Satellite?

Reply:
This is a good question and the solution is still open for discussion. Hopefully we'll flesh that out  and add it to the design soon.

Reply:
Update: When a node talks to another for discovery purposes, it also sends back all of the valid vouchers it has.
If that node does have any trustworthy vouchers, it gets "antechambered".

Reply:
Edit: rather than node discovery, we mean routing table population/refresh.